usb/esp_tinyusb/tusb_cdc_acm.c

```diff
@@ -164,16 +144,24 @@ esp_err_t tinyusb_cdcacm_register_callback(tinyusb_cdcacm_itf_t itf,
     if (acm) {
         switch (event_type) {
         case CDC_EVENT_RX:
```

Comment:
**Refactoring offer** 
Seems that we have 4 equal callbacks and the same logic for them. 
Probably, it can be squashed to callback buffer `tusb_cdcacm_callback_t callback[CDC_EVENT_MAX]` and when we need to call it or change we can do it with a pointer by event_type index. 
This could help to simplify this switch-case structure to: 

    if (event_type < CDC_EVENT_MAX) {
        CDC_ACM_ENTER_CRITICAL();
        acm->callback[event_type] = callback;
        CDC_ACM_EXIT_CRITICAL();
    }
 
Also, the same with unregister callback function.

Reply:
Hi Roman, if you agree, I'd leave this optimization for next round

Reply:
Hey @tore-espressif,

«Later » means « never » 😂.

But yes, no problem.